src/material/checkbox/checkbox.ts

```diff
@@ -203,12 +163,22 @@ export class MatCheckbox
   /** The value attribute of the native input element */
   @Input() value: string;
 
+  /** Whether the checkbox has a ripple. */
+  @Input({transform: booleanAttribute}) disableRipple: boolean;
+
   /** The native `<input type="checkbox">` element */
   @ViewChild('input') _inputElement: ElementRef<HTMLInputElement>;
 
   /** The native `<label>` element */
   @ViewChild('label') _labelElement: ElementRef<HTMLInputElement>;
 
+  /** Tabindex for the checkbox. */
+  @Input({transform: (value: unknown) => (value == null ? undefined : numberAttribute(value))})
```

Comment:
Haven't digged into the transform compiler code, but would the `unknown` here have an effect on type-checking? should it be rather `number|string`?

Also one question, if the transform ends up yielding `undefined`, isn't that non-assignable to `tabIndex: number` with strict null checks enabled?

Reply:
In my testing it also seemed common that apps are passing in `undefined` or `null` to the tabindex to clear it.

Reply:
Yes, makes sense. I'm just wondering, would type checking be incorrect right now and we'd always be able to assign all types of things to the input? e.g. a `RegExp`, an object etc?

Reply:
Right now as in before or after this change? It seems that currently, at least in g3, the type checking for these sorts of properties didn't work correctly because people were assigning all sorts of values to coerced properties.

Reply:
I meant after this change, because previously it was a more more strict type, but now it's just accepting `unknown`. Let's discuss in the comment below as those are basically about the same.